apps/aens/src/aens.erl

```diff
@@ -0,0 +1,78 @@
+%%%-------------------------------------------------------------------
+%%% @copyright (C) 2018, Aeternity Anstalt
+%%% @doc
+%%%    Resolve registered names
+%%% @end
+%%%-------------------------------------------------------------------
+
+-module(aens).
+
+%% API
+-export([resolve/2,
+         get_name_entry/1]).
+
+%%%===================================================================
+%%% Types
+%%%===================================================================
+
+-define(LABEL_SEPARATOR, <<".">>).
+
+%%%===================================================================
+%%% API
+%%%===================================================================
+
+-spec resolve(atom(), binary()) -> {ok, binary()} | {error, atom()}.
+resolve(Type, Binary) ->
+    case lists:reverse(binary:split(Binary, ?LABEL_SEPARATOR, [global, trim])) of
+        [Binary] ->
+            aec_base58c:safe_decode(Type, Binary);
+        [RegistrarNamespace|_Namespace] ->
+            %% When we have more registrars, pick faster and still gov friendly structure
+            %% If we allow to purchase registrars, this will live in regular tree
+            case [RN || RN <- aec_governance:name_registrars(), RN =:= RegistrarNamespace] of
+                [] ->
+                    {error, registrar_unknown};
+                _ ->
+                    {ok, StateTrees} = aec_conductor:top_state_trees(),
+                    NameTree = aec_trees:ns(StateTrees),
+                    case get_name(Binary, NameTree) of
+                        {ok, #{<<"pointers">> := Pointers}} ->
+                            case proplists:get_value(Type, Pointers) of
+                                undefined -> {error, type_not_found};
+                                Val -> {ok, Val}
+                            end;
+                        {error, _} = Err ->
+                            Err
+                    end
+            end
+    end.
+
+-spec get_name_entry(binary()) -> {ok, map()} | {error, atom()}.
+get_name_entry(Name) ->
+    {ok, StateTrees} = aec_conductor:top_state_trees(),
+    NSTree = aec_trees:ns(StateTrees),
+    case get_name(Name, NSTree) of
+        {ok, #{<<"name">>     := Name,
+               <<"name_ttl">> := TTL,
+               <<"pointers">> := Pointers}} ->
+            {ok, #{<<"name">>     => Name,
+                   <<"name_ttl">> => TTL,
+                   <<"pointers">> => jsx:encode(Pointers)}};
```

Comment:
Maybe [Swagger schema](https://github.com/aeternity/epoch/pull/607/files#diff-89b8ec7c2f28f3e3c7d21ca9bf070bb7R223) shall define this field as object/array (rather than string), and JSON encoding delegated to Swagger glue code?